packages/sdk-core/src/modules/attribute/AttributeManager.ts

```diff
@@ -1,13 +1,9 @@
-import { AttributeType } from '../../model/data/BacktraceData';
 import { ReportData } from '../../model/report/ReportData';
 import { BacktraceAttributeProvider } from './BacktraceAttributeProvider';
 import { ReportDataBuilder } from './ReportDataBuilder';
 
 export class AttributeManager {
-    public readonly attributes: Record<string, AttributeType> = {};
-    public readonly annotations: Record<string, unknown> = {};
-
-    private readonly _dynamicAttributeProviders: BacktraceAttributeProvider[] = [];
+    private readonly _attributeSources: Array<Record<string, unknown> | (() => Record<string, unknown>)> = [];
```

Comment:
wouldn't it be better if we will keep all this data as AttributeProvider[]? the new type of the dynamic attribute is already an AttributeProvider with type dynamic. Keeping it in the previous model will be still useful and we will have one interface for everything . Right now we need to verify if something is a function or not

Reply:
Good idea. Changed.